### PR TITLE
18Carolinas: bug fixes; EMR sales of companies

### PR DESCRIPTION
**Pull request by roseundy**
Common file edits:
- View::EmergencyMoney - add ability to sell private companies to the bank
- View::BuyPower - fix must_buy_train to be must_buy_power

### CODE REVIEW

File: lib/engine/game/g_18_carolinas/step/track.rb [hunk 1]
```diff
@@ -62,7 +62,7 @@ def revenue_fail
           end
 
           def process_run_routes(action)
-            hexes = action.routes[0].connection_hexes.flat_map.uniq { |h| @game.hex_by_id(h) }
+            hexes = action.routes[0].connection_hexes.flatten.uniq.map { |h| @game.hex_by_id(h) }.uniq
```
Comment (tobymao) [inline lib/engine/game/g_18_carolinas/step/track.rb:65]: why do you have two uniqs

File: lib/engine/game/g_18_carolinas/game.rb [hunk 2]
```diff
@@ -721,12 +723,20 @@ def train_power?
           true
         end
 
+        def total_emr_buying_power(player, corporation)
+          corporation.cash +
```
Comment (tobymao) [inline lib/engine/game/g_18_carolinas/game.rb:727]: one line

File: lib/engine/game/g_18_carolinas/step/track.rb (hunk 1 above)
Comment (roseundy) [reply lib/engine/game/g_18_carolinas/step/track.rb:65]: Uh, to make sure it's extra unique? I'll fix.

File: lib/engine/game/g_18_carolinas/game.rb (hunk 2 above)
Comment (roseundy) [reply lib/engine/game/g_18_carolinas/game.rb:727]: OK

**Review by tobymao — APPROVED** [review]

**Pull request merged**